tools/wasi/wasi.hpp

```diff
@@ -0,0 +1,25 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <bytes.hpp>
+#include <filesystem>
+#include <iosfwd>
+#include <optional>
+
+namespace fizzy::wasi
+{
+std::optional<bytes> load_wasm_binary(const std::filesystem::path& path, std::ostream& err);
+
+/// Executes WASI main function with given CLI arguments.
+///
+/// @param  argc    Number of CLI arguments.
+/// @param  argv    Array of CLI arguments.
+/// @param  err     Error output stream.
+/// @return         False in case of error.
+///
+/// @todo Make noexcept.
+bool run(int argc, const char* argv[], std::ostream& err);
```

Comment:
Please add a note that path to a binary is expected as first argument. 

Also maybe it should be named `load_and_run`.